lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/SpellCheckerTest.java

```diff
@@ -61,59 +61,74 @@ public void needAffixOnAffixes() throws Exception {
     doTest("needaffix5");
   }
 
+  @Test
   public void testBreak() throws Exception {
     doTest("break");
   }
 
-  public void testBreakDefault() throws Exception {
+  @Test
+  public void breakDefault() throws Exception {
     doTest("breakdefault");
   }
 
-  public void testBreakOff() throws Exception {
+  @Test
+  public void breakOff() throws Exception {
     doTest("breakoff");
   }
 
-  public void testCompoundrule() throws Exception {
+  @Test
+  public void compoundrule() throws Exception {
     doTest("compoundrule");
   }
 
-  public void testCompoundrule2() throws Exception {
+  @Test
+  public void compoundrule2() throws Exception {
     doTest("compoundrule2");
   }
 
-  public void testCompoundrule3() throws Exception {
+  @Test
+  public void compoundrule3() throws Exception {
     doTest("compoundrule3");
   }
 
-  public void testCompoundrule4() throws Exception {
+  @Test
+  public void compoundrule4() throws Exception {
     doTest("compoundrule4");
   }
 
-  public void testCompoundrule5() throws Exception {
+  @Test
+  public void compoundrule5() throws Exception {
     doTest("compoundrule5");
   }
 
-  public void testCompoundrule6() throws Exception {
+  @Test
+  public void compoundrule6() throws Exception {
     doTest("compoundrule6");
   }
 
-  public void testCompoundrule7() throws Exception {
+  @Test
+  public void compoundrule7() throws Exception {
     doTest("compoundrule7");
   }
 
-  public void testCompoundrule8() throws Exception {
+  @Test
+  public void compoundrule8() throws Exception {
     doTest("compoundrule8");
   }
 
-  public void testGermanCompounding() throws Exception {
+  @Test
+  public void germanCompounding() throws Exception {
     doTest("germancompounding");
   }
 
   protected void doTest(String name) throws Exception {
-    InputStream affixStream =
-        Objects.requireNonNull(getClass().getResourceAsStream(name + ".aff"), name);
-    InputStream dictStream =
-        Objects.requireNonNull(getClass().getResourceAsStream(name + ".dic"), name);
+    checkSpellCheckerExpectations(
```

Comment:
When can resources in tests be not in the local file system?
Do you mean passing an URL instead?

Reply:
Is there any utility to read all lines from an input stream conveniently, or should I use the plain old `BufferedReader`?

Reply:
BufferedReader is the way to go!

Reply:
(don't know if any utility method like that exists).

Reply:
URLs aren't very friendly, it's hard to check if they exist (probably with try-catch, but that's ugly). Can resources in tests ever be not in the local file system?

Reply:
Why is it hard? getResource returns null if a resource doesn't exist. getResourceAsStream returns null too.

Reply:
This is now a method reused in the new test which deals with some random path on the disk, unrelated to resources.

Reply:
Well, this isn't too fortunate then. I'll take a look later and maybe come up with more concrete suggestions.

Reply:
You can't really convert resource URLs to paths with url.getPath. This breaks, as I suspected. On Windows you get:
```
java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/Work/apache/lucene/lucene.master/lucene/analysis/common/build/classes/java/test/org/apache/lucene/analysis/hunspell/i53643.aff
   >         at __randomizedtesting.SeedInfo.seed([FE61D482FAEDBB53:CE18D8B46A2785A8]:0)
   >         at java.base/sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
   >         at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
   >         at java.base/sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
   >         at java.base/sun.nio.fs.WindowsPath.parse(WindowsPath.java:92)
   >         at java.base/sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:229)
   >         at java.base/java.nio.file.Path.of(Path.java:147)
```

A better method is to go through the URI - Path.of(url.toUri()). I've modified the code slightly, please take a look.  

Also, can you rename tests to follow TestXXX convention? This may be enforced in the future and will spare somebody some work to rename.

Reply:
Ah... can't push to your repo (there is a checkbox to enable committers to do so - please use it, makes edits easier :). Here is the commit:

https://github.com/dweiss/lucene-solr/commit/618a2d3b5bb51eb0e35322a9c56b97bdce7d728b

Reply:
Thanks for looking into this and for your patch! I've no idea why you can't push, I've got the checkbox enabled on this PR:

![image](https://user-images.githubusercontent.com/122009/106570058-afadb100-6535-11eb-83d4-bd16102f31b5.png)

That's what https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork seems to recommend. I also checked the repository settings and couldn't find an option about committers. I can invite you personally though :)

By renaming do you mean `TestSpellChecker`? I'll do it, thanks for the suggestion, but preferably a bit later, when there won't be so much merging around this class :)

BTW what do you think about renaming `SpellChecker` into `Hunspell`?

Reply:
Mhmm... let me try again then. It's weird - tried last night and got permission denied. Could be that I pulled your changes via https and not ssh... Sorry, it was late.

Reply:
Renaming SpellChecker to Hunspell - yes, I think  it's a good idea. Renaming tests later - absolutely.